### DISCUSSION TITLE
Lack of usage examples

### DISCUSSION THREAD

**Discussion by pchemguy** [rusqlite/rusqlite#1526] [Questions]
The docs contain only a single very basic usage example. Why is that?

**Comment by gwenn** [comment]
> single

https://github.com/search?q=repo%3Arusqlite%2Frusqlite%20%22%60%60%60rust%22&type=code
Are you kidding ?
(and some unit tests which can be considered as examples)

**Reply by pchemguy** [reply]
No, I am not. One thing does is not a substitute for the other. The inline examples illustrate how to use a particular feature to someone who looks for that feature. When I just discovered the library, I have no clue what to look for. And I do not want to start investing my time in learning before seeing working examples in action. That is the point behind `cargo run --example <>`, which you essentially ignore/discard. Then I look inside those examples for common patterns and features, knowing I can copy-paste code, which I just executed.

The whole point behind `cargo run --example <>` command is that as someone

**Comment by gwenn** [comment]
> working examples in action

Unit tests are working examples in action.
And inline examples can be copy/pasted.
For exemple:
https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.open
can be run here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3cb15f11b54779c05cdf22fb686b9cc2

**Comment by gwenn** [comment]
https://github.com/sfackler/rust-postgres/tree/master (no `examples` folder but inline examples)
https://github.com/stainless-steel/sqlite (idem)
It seems to be the norm...
